tracer/src/Datadog.Trace/TracerManager.cs

```diff
@@ -691,22 +691,8 @@ private static void OneTimeSetup(TracerSettings tracerSettings)
             // start the heartbeat loop
             _heartbeatTimer = new Timer(HeartbeatCallback, state: null, dueTime: TimeSpan.Zero, period: TimeSpan.FromMinutes(1));
 
-            if (FrameworkDescription.Instance.OSPlatform == OSPlatformName.Linux && Environment.Is64BitProcess && !Util.EnvironmentHelpers.IsServerlessEnvironment())
```

Comment:
This code was moved as-is into `ServiceDiscoveryHelpers` and then refactored to make it easier to test